pilot/pkg/config/memory/controller.go

```diff
@@ -26,9 +26,35 @@ import (
 	"istio.io/istio/pkg/config/schema/collection"
 )
 
+type Config struct {
+	HasSynced func() bool
+	SyncProc  bool
+}
+
 type controller struct {
 	monitor     Monitor
 	configStore model.ConfigStore
+	opts        *Config
+}
+
+// NewControllerWithOptions return an implementation of model.ConfigStoreCache.
+func NewControllerWithOptions(cs model.ConfigStore, opts *Config) model.ConfigStoreCache {
+	if opts == nil {
+		opts = &Config{}
+	}
+
+	out := &controller{
+		configStore: cs,
+		opts:        opts,
+	}
+
+	if opts.SyncProc {
```

Comment:
It looks no better than before

Reply:
just keep a compability with existing usage.